src/mapeo-manager.js

```diff
@@ -103,6 +115,25 @@ export class MapeoManager {
     } else {
       this.#coreStorage = coreStorage
     }
+
+    this.#mediaServer = fastify({ logger: mediaServerOpts?.logger })
+
+    this.#mediaServer.register(BlobServerPlugin, {
+      prefix: MEDIA_SERVER_BLOBS_PREFIX,
+      getBlobStore: async (projectPublicId) => {
+        const project = await this.getProject(projectPublicId)
+        return project[kBlobStore]
+      },
+    })
+
+    this.#deferredMediaServerListen = pDefer()
```

Comment:
I think maybe move MediaServer into its own class (since we're also going to add the icon plugin, and possible more in the future), and look at https://github.com/digidem/start-stop-state-machine to help with managing start-stop state, then you can await mediaServer.started() (we will need to start and stop listening on a port when the app is put in the background - so the `mapeoManager.start()` and `mapeoManager.stop()` will need to be exposed, at least to the backend process (ok to expose these to the client too I think). I'll need to attach DNS-SD to these start and stop methods too.

Side-note about this: with your code as-written you would need a `this.#deferredMediaServerListen.promise.catch(noop)` to avoid an uncaught error when you reject this.